includes/multi-currency/Analytics.php

```diff
@@ -0,0 +1,266 @@
+			'product_gross_revenue' => 'CASE WHEN wcpay_multicurrency_default_currency_postmeta.meta_value IS NOT NULL THEN product_gross_revenue * (1 / wcpay_multicurrency_exchange_rate_postmeta.meta_value ) ELSE product_gross_revenue END',
+		],
+		'variations' => [
+			'product_net_revenue'   => 'CASE WHEN wcpay_multicurrency_default_currency_postmeta.meta_value IS NOT NULL THEN product_net_revenue * (1 / wcpay_multicurrency_exchange_rate_postmeta.meta_value) ELSE product_net_revenue END',
+			'product_gross_revenue' => 'CASE WHEN wcpay_multicurrency_default_currency_postmeta.meta_value IS NOT NULL THEN product_gross_revenue * (1 / wcpay_multicurrency_exchange_rate_postmeta.meta_value ) ELSE product_gross_revenue END',
+		],
+		'categories' => [
+			'product_net_revenue'   => 'CASE WHEN wcpay_multicurrency_default_currency_postmeta.meta_value IS NOT NULL THEN product_net_revenue * (1 / wcpay_multicurrency_exchange_rate_postmeta.meta_value) ELSE product_net_revenue END',
+			'product_gross_revenue' => 'CASE WHEN wcpay_multicurrency_default_currency_postmeta.meta_value IS NOT NULL THEN product_gross_revenue * (1 / wcpay_multicurrency_exchange_rate_postmeta.meta_value ) ELSE product_gross_revenue END',
+		],
+		'taxes'      => [
+			'SUM(total_tax)'    => 'SUM(CASE WHEN wcpay_multicurrency_default_currency_postmeta.meta_value IS NOT NULL THEN total_tax * (1 / wcpay_multicurrency_exchange_rate_postmeta.meta_value ) ELSE total_tax END)',
+			'SUM(order_tax)'    => 'SUM(CASE WHEN wcpay_multicurrency_default_currency_postmeta.meta_value IS NOT NULL THEN order_tax * (1 / wcpay_multicurrency_exchange_rate_postmeta.meta_value ) ELSE order_tax END)',
+			'SUM(shipping_tax)' => 'SUM(CASE WHEN wcpay_multicurrency_default_currency_postmeta.meta_value IS NOT NULL THEN shipping_tax * (1 / wcpay_multicurrency_exchange_rate_postmeta.meta_value ) ELSE shipping_tax END)',
+		],
+		'coupons'    => [
+			'discount_amount' => 'CASE WHEN wcpay_multicurrency_default_currency_postmeta.meta_value IS NOT NULL THEN discount_amount * (1 / wcpay_multicurrency_exchange_rate_postmeta.meta_value ) ELSE discount_amount END',
+		],
+	];
+
+	/**
+	 * Instance of MultiCurrency.
+	 *
+	 * @var MultiCurrency $multi_currency
+	 */
+	private $multi_currency;
+
+	/**
+	 * Constructor
+	 *
+	 * @param MultiCurrency $multi_currency Instance of MultiCurrency.
+	 */
+	public function __construct( MultiCurrency $multi_currency ) {
+		$this->multi_currency = $multi_currency;
+		$this->init();
+	}
+
+	/**
+	 * Initialise all actions, filters and hooks related to analytics support.
+	 *
+	 * @return void
+	 */
+	public function init() {
+		if ( is_admin() ) {
+			add_filter( 'admin_enqueue_scripts', [ $this, 'enqueue_admin_scripts' ] );
+		}
+
+		// TODO: Remove this or make it only apply in dev mode.
+		add_filter( 'woocommerce_analytics_report_should_use_cache', [ $this, 'disable_report_caching' ] );
```

Comment:
TODO: Either remove this, or make it based on a filter or a config variable

Reply:
@dmallory42 is this TODO addressed? I see that you added the filter

Reply:
Addressed now, thanks!